File: monitorlibrary/lib/data/content.dart
import 'package:meta/meta.dart';
import 'package:monitorlibrary/data/position.dart';

class Content {
  String? url;
  String? userId, created;
  Position? position;
  Content(
      {required this.url,
      required this.userId,
      required this.position,
      this.created});

  Content.fromJson(Map data) {
    this.url = data['url'];
    this.userId = data['userId'];
    this.created = data['created'];
    if (data['position'] != null) {
      this.position = Position.fromJson(data['position']);
    }
  }
  
  Map<String, dynamic> toJson() {
    Map pos = Map();
    if (position != null) {
      pos = position!.toJson();
    }
    Map<String, dynamic> map = {
      'userId': userId,
      'url': url,
      'created': created,
      'position': pos,
    };
    return map;
  }
}

File: webapi-spring/src/main/java/com/monitorz/webapi/data/Content.java
package com.monitorz.webapi.data;

public class Content {
    private String url;
    private String userId, created;
    private Position position;

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getCreated() {
        return created;
    }

    public void setCreated(String created) {
        this.created = created;
    }

    public Position getPosition() {
        return position;
    }

    public void setPosition(Position position) {
        this.position = position;
    }
}
